ddl/ddl_api.go

```diff
@@ -7668,6 +7668,20 @@ func (d *ddl) DropResourceGroup(ctx sessionctx.Context, stmt *ast.DropResourceGr
 		return err
 	}
 
+	// check to see if some user has dependency on the group
+	exec := ctx.(sqlexec.RestrictedSQLExecutor)
+	internalCtx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL)
+	sql := "select count(*) from mysql.user where lower(json_extract(user_attributes, '$.resource_group')) ='\"" + groupName.L + "\"'"
```

Comment:
- Do we have a better way without the SQL? cc @tiancaiamao 
- This way cannot cover the case: 
    - admin bind `user1`  to `group1`,  then `user1` running the workload within a session, admin bind `user1` to default resource group but the session still running and consume `group1`, and then admin drop the resource group. finally, the session may meet error because the group is dropped.

Reply:
1. It's a common practice for DDL so far. Another way could be traversing User of MySQLPrivilege.
2. Active long session won't fail but it will fall to use `default` resource group I think.

Reply:
>Active long session won't fail but it will fall to use default resource group I think.
Will the session change the `ResourceGroupName` in the session context after the resource group dropped?

Reply:
No. The session associate with resource name only at session establishment.

Reply:
As discussed offline, I will follow the problem via this issue https://github.com/pingcap/tidb/issues/40724